src/internal_events/http_scrape.rs

```diff
@@ -0,0 +1,90 @@
+use metrics::counter;
+use vector_core::internal_event::InternalEvent;
+
+use super::prelude::{error_stage, error_type, http_error_code};
+
+#[derive(Debug)]
+pub struct HttpScrapeEventsReceived {
+    pub byte_size: usize,
+    pub count: usize,
+    pub uri: http::Uri,
+}
+
+impl InternalEvent for HttpScrapeEventsReceived {
+    fn emit(self) {
+        trace!(
+            message = "Events received.",
+            count = %self.count,
+            byte_size = %self.byte_size,
+            uri = %self.uri,
```

Comment:
I _think_ we'd want to try and get these all to be `&str`s? Also curious if we need the entire `uri` and not just a portion of it.

Ref: https://github.com/vectordotdev/vector/pull/13749/files

Reply:
🤔  Not 100% following on the `&str`s line of thought, can you clarify? The count and byte_size are both of usize type. , Are you referring to only the uri ?

Reply:
Sorry - unclear in my comment. Using `&str` instead of `String` should save on allocations for all the uris

Reply:
Ah, got it. Good call. Also, I will update the member var name to be `url` , since its the full url. These bits were ported direct from the prometheus_scrape code.

Reply:
So while I was attempting to address this I realized that in the PR you linked - there is a critical difference from this code. In that PR, the as_str() function was manually implemented for that enum, and returns a static string.
The counter! / trace! etc macros require a static string, so I can't use a reference there, but I will store a `String` as the member var, then it reduces one more copy for each of the structs.

Reply:
Addressed in https://github.com/vectordotdev/vector/pull/13793/commits/5f7c44231f801eb5ee0be2f2a3549efb6c908e8c